campaign-launcher/client/src/hooks/useStake.ts

```diff
@@ -60,37 +47,29 @@ export const useStake = () => {
   };
 
   const resetData = () => {
-    setStakingData(null);
     setStakingClient(null);
   };
 
-  const fetchStakingData = async (stakingClient: StakingClient) => {
+  const fetchStakingData = async () => {
     checkSupportedChain();
-    try {
-      const stakingInfo = await stakingClient.getStakerInfo(activeAddress!);
-      setStakingData(stakingInfo);
-      return stakingInfo;
-    } catch (error) {
-      console.error('Error fetching staking data', error);
-      return null;
-    }
-  };
-
-  const refetchStakingData = async () => {
     if (stakingClient && activeAddress) {
-      setIsRefetching(true);
-      const info = await fetchStakingData(stakingClient);
-      setIsRefetching(false);
-      return formatTokenAmount(Number(info?.stakedAmount) || 0);
+      setIsFetching(true);
+      try {
+        const stakingInfo = await stakingClient.getStakerInfo(activeAddress);
+        return formatTokenAmount(Number(stakingInfo?.stakedAmount) || 0)
+      } catch (error) {
+        console.error('Error fetching staking data', error);
+        return null;
+      } finally {
+        setIsFetching(false);
+      }
     }
     return 0;
   };
 
   return {
-    stakedAmount: formatTokenAmount(Number(stakingData?.stakedAmount) || 0),
-    refetchStakingData,
+    fetchStakingData,
```

Comment:
Nice simplification

Reply:
most likely, these changes will be reverted